esphome/components/fan/__init__.py

```diff
@@ -220,7 +222,10 @@ async def fan_turn_on_to_code(config, action_id, template_arg, args):
 @automation.register_action("fan.cycle_speed", CycleSpeedAction, FAN_ACTION_SCHEMA)
```

Comment:
Thois should just extend the action schema as its only used in this action
```suggestion
@automation.register_action(
    "fan.cycle_speed", 
    CycleSpeedAction, 
    FAN_ACTION_SCHEMA.extend({
        cv.Optional(CONF_OFF_SPEED_CYCLE, default=True): cv.boolean,
    }),
)
```